pkg/resolution/resolver/git/resolver_test.go

```diff
@@ -765,3 +780,27 @@ func toParams(m map[string]string) []pipelinev1beta1.Param {
 
 	return params
 }
+
+func revisionToCommitSHA(reqURL string, useNthCommit int, revision, defaultRevision, specificCommit string, commitsByBranch map[string][]string, commitByTag map[string]string) string {
+	switch {
+	case useNthCommit > 0:
+		return commitsByBranch[plumbing.Master.Short()][useNthCommit]
+	case specificCommit != "":
+		return specificCommit
+	case revision == "" && reqURL != "":
+		return revisionByBranchOrTag(commitsByBranch, commitByTag, plumbing.Master.Short())
```

Comment:
This has me a little worried, since this could be defaulting to branch names. Instead of trying to plumb all of the repo data through commitByBranch / commitByTag, could we have createTestRepo return the https://pkg.go.dev/github.com/go-git/go-git/v5#Repository and we can use https://pkg.go.dev/github.com/go-git/go-git/v5#Repository.ResolveRevision instead?

Reply:
Good point. I've changed createTestRepo as suggested. PTAL. Thank you!!

Reply:
Discussed with @wlynch offline. By this approach, we seem to compare value returned by [ResolveRevision](https://pkg.go.dev/github.com/go-git/go-git/v5#Repository.ResolveRevision) in test with the value returned by the same function in implementation code which uses same function for anonymous cloning.

Here is the latest update:
- for anonymous cloning: 
I changed createTestRepo to return a slice of commit SHAs. In the test table, I added a new field `expectedNthCommit` so that we can compare the commit SHA.
- for scm api cloning:
The implementation code sets the the digest to be the content sha atm, but we need to change to actual commit sha. See the discussion [here](https://github.com/tektoncd/pipeline/pull/5664#discussion_r1011858087)

wdyt?